examples/diff.c

```diff
@@ -13,6 +13,7 @@
  */
 
 #include "common.h"
+#include "buffer.h"
```

Comment:
This causes CI to fail:
```
In file included from /src/src/common.h:182:0,
                 from /src/src/buffer.h:10,
                 from /src/examples/diff.c:16:
/src/src/util.h:23:0: error: "ARRAY_SIZE" redefined [-Werror]
 #define ARRAY_SIZE(x) (sizeof(x)/sizeof(x[0]))
 ^
In file included from /src/examples/diff.c:15:0:
/src/examples/common.h:34:0: note: this is the location of the previous definition
 #define ARRAY_SIZE(x) (sizeof(x)/sizeof(*x))
 ^
```

I don't think it's needed, now that the `git_buf__initbuf` is gone.